aws/resource_aws_route53_zone.go

```diff
@@ -29,10 +29,11 @@ func resourceAwsRoute53Zone() *schema.Resource {
 
 		Schema: map[string]*schema.Schema{
 			"name": {
-				Type:             schema.TypeString,
-				Required:         true,
-				ForceNew:         true,
-				DiffSuppressFunc: suppressRoute53ZoneNameWithTrailingDot,
```

Comment:
If we see confusing behavior in the inverse direction here with respect to "provider produced inconsistent results after apply", we can implement a `StateFunc` similar to the ones we're removing in the other resources here to allow operators to declare the configuration with a trailing period, but have Terraform correctly know it will be removed, e.g.

```go
StateFunc: func(v interface{}) string {
  return strings.TrimSuffix(v.(string), ".")
},
```

Reply:
makes sense - i'll add it to the data-source as well since that originally accounted for the trailing period to be in the input as well